modules/afsocket/transport-mapper-inet.c

```diff
@@ -544,6 +559,22 @@ transport_mapper_syslog_apply_transport(TransportMapper *s, GlobalConfig *cfg)
       self->delegate_tls_start_to_logproto = TRUE;
       self->super.transport_name = g_strdup("rfc5425+proxied-tls-passthrough");
     }
+  else if (strcasecmp(transport, "auto") == 0)
+    {
+      /* NOTE: this is temporary until TLS auto detection does not exist.
+       * In this case transport(auto) + tls() configuration means we want to
+       * start TLS first and then do the framing auto-detection.  If we have
+       * TLS auto-detection this entire block can be removed, as the "auto"
+       * proto would detect TLS and start it if needed.
```

Comment:
TLS auto-detection sounds like a security issue for me. TLS provides encryption and authentication.
If we want to allow non-TLS and TLS traffic on the same port, we should notify the user with a big warning that neither encryption nor authentication is guaranteed. We could probably do this when `peer-verify(off)` is set, but even then we should warn the user that encryption is also optional so vulnerable to "downgrade attack".